### PR TITLE
Remove React.creatClass 

### PR DESCRIPTION
**Pull request by mlecoq**
Hi, 

I have removed version update, reviewed bindings and run prettier (I have upgraded it, the version used was failing ...)

### CODE REVIEW

**Comment by mlecoq** [comment]
@maisano have you reviewed my pull request ? Something is wrong ?

**Comment by thinq4yourself** [comment]
Bump on this being merged in.

File: src/RouteTransition.jsx [hunk 1]
```diff
@@ -51,20 +35,22 @@ const RouteTransition = React.createClass({
...
+        data: this.props.children,
+        style: ensureSpring(this.props.atActive)
+      }
+    ];
+  }
 
   willEnter() {
-    return this.props.atEnter;
-  },
+    return () => this.props.atEnter;
```
Comment (maisano) [inline src/RouteTransition.jsx]: i would prefer we don't add extra functions and just bind these reference in the constructor. can you update that?

File: src/RouteTransition.jsx [hunk 2]
```diff
@@ -75,28 +61,46 @@ const RouteTransition = React.createClass({
...
       ? createElement(this.props.component, props, config.data)
       : cloneElement(config.data, props);
-  },
+  }
 
   renderRoutes(interpolatedStyles) {
     return (
       <div className={this.props.className} style={this.props.style}>
-        {interpolatedStyles.map(this.renderRoute)}
+        {interpolatedStyles.map(config => this.renderRoute(config))}
```
Comment (maisano) [inline src/RouteTransition.jsx]: same as mentioned above here.

File: src/RouteTransition.jsx (hunk 1 above)
Comment (mlecoq) [inline src/RouteTransition.jsx]: Done !

File: src/RouteTransition.jsx (hunk 2 above)
Comment (mlecoq) [inline src/RouteTransition.jsx]: Done !

**Comment by maisano** [comment]
thanks @mlecoq!

**Pull request merged**